### ISSUE TITLE
easyeffects killed by DBUS without any reason

### ISSUE THREAD

**Issue by jpVm5jYYRE1VIKL** [wwmm/easyeffects#3450]
### EasyEffects Version

7.1.8

### What package are you using?

Other (specify below)

### Distribution

manjaro

### Describe the bug

normally using easyeffects and  random moments it just close. 

### Expected Behavior

must work

### Debug Log

<details><summary>Debug Log</summary><br/>

```
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.925:  pipe_manager.cpp:1039   new metadata property: 204, target.node, Spa:Id, 140
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.925:  pipe_manager.cpp:1039   new metadata property: 204, target.object, Spa:Id, 1290
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.935:  app_info.cpp:377         disposed
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.936:  app_info.cpp:385         finalized
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.936:  app_info.cpp:63 data struct destroyed
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.936:  node_info_holder.cpp:104        204, telegram-desktop finalized
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.937:  pipe_manager.cpp:1039   new metadata property: 204, target.node, , 
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.937:  pipe_manager.cpp:1039   new metadata property: 204, target.object, , 
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.945:  pipe_manager.cpp:1346   No node with id 204 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.945:  pipe_manager.cpp:1346   No node with id 204 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.962:  output_level.cpp:45     soe: output_level: PipeWire blocksize: 512
(easyeffects:124515): easyeffects-DEBUG: 16:54:06.962:  output_level.cpp:46     soe: output_level: PipeWire sampling rate: 48000
(easyeffects:124515): easyeffects-DEBUG: 16:54:07.125:  stream_input_effects.cpp:189    No app linked to our device wants to play. Unlinking our filters.
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.905:  pipe_manager.cpp:1039   new metadata property: 0, default.configured.audio.sink, Spa:String:JSON, { "name": "alsa_output.pci-0000_00_1f.3.analog-stereo" }
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.925:  pipe_manager.cpp:1039   new metadata property: 0, default.audio.sink, Spa:String:JSON, {"name":"alsa_output.pci-0000_00_1f.3.analog-stereo"}
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.925:  application.cpp:104     new default output device: alsa_output.pci-0000_00_1f.3.analog-stereo
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.925:  pipe_manager.cpp:1346   No node with id 204 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.925:  pipe_manager.cpp:1346   No node with id 204 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.925:  stream_output_effects.cpp:302   disconnecting the autogain filter from PipeWire
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.930:  output_level.cpp:45     soe: output_level: PipeWire blocksize: 1024
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.930:  output_level.cpp:46     soe: output_level: PipeWire sampling rate: 48000
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.930:  pipe_manager.cpp:265     167 ee_soe_autogain has been removed
(easyeffects:124515): easyeffects-DEBUG: 16:54:10.930:  stream_output_effects.cpp:302   disconnecting the limiter filter from PipeWire
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.163:  pipe_manager.cpp:265     108 ee_soe_limiter has been removed
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.173:  plugin_base.cpp:406     soe: autogain successfully connected to PipeWire graph
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.174:  pipe_manager.cpp:1343   easyeffects_sink port 86 is connected to ee_soe_autogain port 141
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.175:  pipe_manager.cpp:1343   easyeffects_sink port 109 is connected to ee_soe_autogain port 129
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.179:  plugin_base.cpp:406     soe: limiter successfully connected to PipeWire graph
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.180:  pipe_manager.cpp:1343   ee_soe_autogain port 117 is connected to ee_soe_limiter port 134
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.181:  pipe_manager.cpp:1343   ee_soe_autogain port 108 is connected to ee_soe_limiter port 87
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.182:  pipe_manager.cpp:1346   No node with id 137 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.184:  pipe_manager.cpp:1346   No node with id 137 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.185:  pipe_manager.cpp:1346   No node with id 115 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.187:  pipe_manager.cpp:1346   No node with id 115 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.187:  pipe_manager.cpp:1346   No node with id 115 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.188:  pipe_manager.cpp:1346   No node with id 115 in our node_map
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.191:  output_level.cpp:45     soe: output_level: PipeWire blocksize: 512
(easyeffects:124515): easyeffects-DEBUG: 16:54:11.191:  output_level.cpp:46     soe: output_level: PipeWire sampling rate: 48000
Killed

```
</details>

### Additional Information

in journalctl it show soemthing like this : 

```
11/10/2024 16:34	pipewire	pw.node: (alsa_output.pci-0000_00_1f.3.analog-stereo-60) graph xrun not-triggered (3 suppressed)
11/10/2024 16:34	pipewire	pw.node: (alsa_output.pci-0000_00_1f.3.analog-stereo-60) xrun state:0x7f7a7a27e008 pending:3/10 s:71888861218413 a:71888869360798 f:71888869371897 waiting:8142385 process:11099 status:triggered
11/10/2024 16:34	pipewire	pw.node: (ee_soe_limiter-130) xrun state:0x7f7a79b8e008 pending:0/2 s:71888882527676 a:71888882528914 f:71888869347224 waiting:1238 process:18446744073696369926 status:awake
11/10/2024 16:34	systemd	dbus-:1.2-com.github.wwmm.easyeffects@13.service: Main process exited, code=killed, status=9/KILL
11/10/2024 16:34	systemd	dbus-:1.2-com.github.wwmm.easyeffects@13.service: Failed with result 'signal'.
11/10/2024 16:34	systemd	dbus-:1.2-com.github.wwmm.easyeffects@13.service: Consumed 24min 51.550s CPU time, 132.8M memory peak.

```

**Comment by jpVm5jYYRE1VIKL** [comment]
After some investigations i can easily recreate issue using telegram-desktop application : 

easyeffect killed by systemd after pause video and start video playback again .

```
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.687:  output_level.cpp:46     soe: output_level: PipeWire sampling rate: 48000
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.756:  pipe_manager.cpp:1303   Stream/Output/Audio 148 telegram-desktop with serial 3629 has been added
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.756:  pipe_manager.cpp:360    The output stream telegram-desktop does not have as target the same output device used as EE: alsa_output.pci-0000_00_1f.3.analog-stereo
 The user wants it to play to device 43. We will ignore this stream.
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.756:  pipe_manager.cpp:424    Stream/Output/Audio 148 telegram-desktop has been removed
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.757:  pipe_manager.cpp:1039   new metadata property: 148, target.node, Spa:Id, 157
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.757:  pipe_manager.cpp:1039   new metadata property: 148, target.object, Spa:Id, 3510
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.759:  app_info.cpp:377         disposed
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.760:  app_info.cpp:385         finalized
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.760:  app_info.cpp:63 data struct destroyed
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.760:  node_info_holder.cpp:104        148, telegram-desktop finalized
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.761:  pipe_manager.cpp:1039   new metadata property: 148, target.node, , 
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.761:  pipe_manager.cpp:1039   new metadata property: 148, target.object, , 
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.762:  pipe_manager.cpp:1346   No node with id 148 in our node_map
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.762:  pipe_manager.cpp:1346   No node with id 148 in our node_map
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.765:  pipe_manager.cpp:1346   No node with id 148 in our node_map
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.765:  pipe_manager.cpp:1346   No node with id 148 in our node_map
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.771:  output_level.cpp:45     soe: output_level: PipeWire blocksize: 512
(easyeffects:136827): easyeffects-DEBUG: 20:15:44.771:  output_level.cpp:46     soe: output_level: PipeWire sampling rate: 48000
Killed
```

**Comment by wwmm** [comment]
What is the output of `chrt -ap pid` where `pid` is EasyEffects pid number. Often the reason why the system kills EAsyEffects is because services like ananicy had a policy setting all EasyEffects threads to realtime priority. Something that should never be done.

**Comment by jpVm5jYYRE1VIKL** [comment]
> What is the output of `chrt -ap pid` where `pid` is EasyEffects pid number. 

Hi thanks for your reply . This is output from easyeffects :

```
pid 154370's current scheduling policy: SCHED_OTHER
pid 154370's current scheduling priority: 0
pid 154371's current scheduling policy: SCHED_OTHER
pid 154371's current scheduling priority: 0
pid 154372's current scheduling policy: SCHED_OTHER
pid 154372's current scheduling priority: 0
pid 154373's current scheduling policy: SCHED_OTHER
pid 154373's current scheduling priority: 0
pid 154374's current scheduling policy: SCHED_OTHER
pid 154374's current scheduling priority: 0
pid 154375's current scheduling policy: SCHED_OTHER
pid 154375's current scheduling priority: 0
pid 154377's current scheduling policy: SCHED_OTHER
pid 154377's current scheduling priority: 0
pid 154378's current scheduling policy: SCHED_OTHER
pid 154378's current scheduling priority: 0
pid 154379's current scheduling policy: SCHED_RR|SCHED_RESET_ON_FORK
pid 154379's current scheduling priority: 20
 
```

**Comment by wwmm** [comment]
The scheduling priorities are fine. So that is not the source of problem. Which plugins are you using in the effects pipeline? Maybe the system was not fast enough to handle them in the realtime thread and the system killed EasyEffects. But it is hard to see this happening in a recent CPU.

Here on my Arch Linux installation I've never seen the system killing EasyEffects. So there must be something different on Manjaro triggering this. The question is what.

**Comment by jpVm5jYYRE1VIKL** [comment]
i using autogain(MS)->limiter . Well i have i7-14700KF i think it is more than enough. Especisally that it is idle andnot have any another tasks. 

PS: i know about modern issues with core i7 but i almost sure that this is not an issue.

**Comment by wwmm** [comment]
> i using autogain(MS)->limiter . Well i have i7-14700KF i think it is more than enough. Especisally that it is idle andnot have any another tasks.
> 
> PS: i know about modern issues with core i7 but i almost sure that this is not an issue.

This CPU should not have issues with any of the plugins in EasyEffects. I wish the system printed the reason why it kills the process. Some weeks ago several users were having a similar issue but in that case the problem was the presence of ananicy rules that were doing  something they should not to do to EasyEffects priorities. But this isn't the case here. Based on the kill message a memory leak also does not seem to be the case =/

**Comment by jpVm5jYYRE1VIKL** [comment]
Today got bit more details : 

During youtube playback got core : 

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007ffa6ceb296c in ?? () from /usr/lib/libc.so.6
[Current thread is 1 (Thread 0x7ffa66dfe6c0 (LWP 49790))]
(gdb) bt 
#0  0x00007ffa6ceb296c in ?? () from /usr/lib/libc.so.6
#1  0x00007ffa6ceb3902 in ?? () from /usr/lib/libc.so.6
#2  0x00007ffa6ceb4906 in realloc () from /usr/lib/libc.so.6
#3  0x00007ffa6f46fc1b in g_realloc () from /usr/lib/libglib-2.0.so.0
#4  0x00007ffa6f44f29f in ?? () from /usr/lib/libglib-2.0.so.0
#5  0x00007ffa6f44fa49 in ?? () from /usr/lib/libglib-2.0.so.0
#6  0x00007ffa6f454b47 in ?? () from /usr/lib/libglib-2.0.so.0
#7  0x00007ffa6f455107 in g_hash_table_add () from /usr/lib/libglib-2.0.so.0
#8  0x00007ffa6f468900 in ?? () from /usr/lib/libglib-2.0.so.0
#9  0x00007ffa6f468ab8 in g_source_attach () from /usr/lib/libglib-2.0.so.0
#10 0x00007ffa6f469db7 in g_main_context_invoke_full () from /usr/lib/libglib-2.0.so.0
#11 0x00007ffa6e5e4251 in ?? () from /usr/lib/libgio-2.0.so.0
#12 0x00007ffa67e4982f in ?? () from /usr/lib/gio/modules/libdconfsettings.so
#13 0x00007ffa6e5fba39 in ?? () from /usr/lib/libgio-2.0.so.0
#14 0x00007ffa6f469299 in ?? () from /usr/lib/libglib-2.0.so.0
#15 0x00007ffa6f4cbec7 in ?? () from /usr/lib/libglib-2.0.so.0
#16 0x00007ffa6f468795 in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0
#17 0x00007ffa67e462fe in ?? () from /usr/lib/gio/modules/libdconfsettings.so
#18 0x00007ffa6f4991b6 in ?? () from /usr/lib/libglib-2.0.so.0
#19 0x00007ffa6cea339d in ?? () from /usr/lib/libc.so.6
#20 0x00007ffa6cf2849c in ?? () from /usr/lib/libc.so.6
```

**Comment by wwmm** [comment]
>During youtube playback got core :
>PS: sorry after some investigation i found that this is another issue.

Maybe these issues are related. Problems in glib can hit EasyEffects depending on what it is.

**Comment by jpVm5jYYRE1VIKL** [comment]
hi. In my original issue that systemd send kill signal to easyeffect it is not crashing. Core dump happend only once before it never happend  . Can it be that in a moment when you register service with dbus some parameters wrong installated and systemd detect it as bogus or zombie and send kill ? Is it possible somehow unbind from systemd relation ?

**Comment by wwmm** [comment]
> Can it be that in a moment when you register service with dbus some parameters wrong installated and systemd detect it as bogus or zombie and send kill ? Is it possible somehow unbind from systemd relation ?

As the registration isn't done by EasyEffects code but by glib/gtk even if this is the source of the problem it can not be fixed on our side.

>In case if i attach to easyeffects with strace using : strace -p -ff -o sss i cannot recreate issue anymore.
I E with strace attached easyeffects not killed . Without almost immediately killed .

Hum... I can understand this making a difference when there is a segmentation fault caused by thread racing. Running through strace will probably change the timing between threads events. But I do not see how this can make the system to not kill the process. Usually taking too long using critical resources is what makes the system kill a process.

**Comment by jpVm5jYYRE1VIKL** [comment]
Is it possible somehow to disable autokill ? 
. 

i see that most often EE killed if exist "output_level.cpp:45     soe: output_level: PipeWire blocksize: 512" if blocksize bigger kill almost never happens.
I am also configured auditd to find what process kill it. Auditd to do detect kill signal to EE. but if i send KILL -9 manually then auditddetect it. Weird issue.

extra trace
what was interesting with this one . I watched video after i finished video i just closed firefox. Immediately after it killed easyeffects

```
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:137     input autoloading: skip "plasma_sink" candidate since it does not match "alsa_input.usb-GENERAL_GENERAL_WEBCAM_JH0319_20200710_v012-02.mono-fallback" input device
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:137     input autoloading: skip "alsa_output.pci-0000_00_1f.3.analog-stereo" candidate since it does not match "alsa_input.usb-GENERAL_GENERAL_WEBCAM_JH0319_20200710_v012-02.mono-fallback" input device
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:137     input autoloading: skip "alsa_input.pci-0000_00_1f.3.analog-stereo" candidate since it does not match "alsa_input.usb-GENERAL_GENERAL_WEBCAM_JH0319_20200710_v012-02.mono-fallback" input device
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:131     input autoloading: target node "alsa_input.usb-GENERAL_GENERAL_WEBCAM_JH0319_20200710_v012-02.mono-fallback" matches the input device name
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:150     output autoloading: device "alsa_card.pci-0000_00_1f.3" has changed its output route to "analog-output-lineout"
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:163     output autoloading: skip "plasma_source" candidate since it does not match "alsa_output.pci-0000_00_1f.3.analog-stereo" output device
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:163     output autoloading: skip "plasma_sink" candidate since it does not match "alsa_output.pci-0000_00_1f.3.analog-stereo" output device
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.497:         application.cpp:157     output autoloading: target node "alsa_output.pci-0000_00_1f.3.analog-stereo" matches the output device name
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.499:         pipe_manager.cpp:1343   Firefox port 158 is connected to easyeffects_sink port 210
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.499:         pipe_manager.cpp:1343   Firefox port 180 is connected to easyeffects_sink port 79
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.500:         pipe_manager.cpp:1343   Firefox port 105 is connected to easyeffects_sink port 210
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.500:         pipe_manager.cpp:1343   Firefox port 100 is connected to easyeffects_sink port 79
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.509:         limiter.cpp:161 soe: limiter latency: 0.0054166666 s
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.509:         output_level.cpp:45     soe: output_level: PipeWire blocksize: 512
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.509:         output_level.cpp:46     soe: output_level: PipeWire sampling rate: 48000
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.524:         app_info.cpp:178        cannot lookup application icon plasma_source in /usr/local/share/pixmaps
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.524:         app_info.cpp:178        cannot lookup application icon plasma_source in /usr/local/share/pixmaps
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.524:         app_info.cpp:178        cannot lookup application icon plasma_source in /usr/local/share/pixmaps
(easyeffects:1024879): easyeffects-DEBUG: 20:05:45.525:         effects_base.cpp:311    soe: pipeline latency: 5.4166665 ms
(easyeffects:1024879): easyeffects-DEBUG: 20:05:55.888:         stream_input_effects.cpp:189    No app linked to our device wants to play. Unlinking our filters.
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.540:         pipe_manager.cpp:265    Stream/Output/Audio 240 Firefox has been removed
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.540:         pipe_manager.cpp:1039   new metadata property: 240, , , 
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.541:         app_info.cpp:377        Firefox disposed
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.541:         app_info.cpp:385        Firefox finalized
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.541:         app_info.cpp:63 data struct destroyed
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.541:         node_info_holder.cpp:104        240, Firefox finalized
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.610:         pipe_manager.cpp:1303   Stream/Output/Audio 240 Firefox with serial 21182 has been added
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.610:         pipe_manager.cpp:1039   new metadata property: 240, target.node, Spa:Id, 67
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.610:         pipe_manager.cpp:1039   new metadata property: 240, target.object, Spa:Id, 21062
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.617:         pipe_manager.cpp:1343   Firefox port 180 is connected to easyeffects_sink port 210
(easyeffects:1024879): easyeffects-DEBUG: 20:06:00.617:         pipe_manager.cpp:1343   Firefox port 207 is connected to easyeffects_sink port 79
(easyeffects:1024879): easyeffects-DEBUG: 20:08:34.861:         pipe_manager.cpp:265    Stream/Output/Audio 240 Firefox has been removed
(easyeffects:1024879): easyeffects-DEBUG: 20:08:34.862:         pipe_manager.cpp:1039   new metadata property: 240, , , 
(easyeffects:1024879): easyeffects-DEBUG: 20:08:34.862:         app_info.cpp:377        Firefox disposed
(easyeffects:1024879): easyeffects-DEBUG: 20:08:34.862:         app_info.cpp:385        Firefox finalized
(easyeffects:1024879): easyeffects-DEBUG: 20:08:34.862:         app_info.cpp:63 data struct destroyed
(easyeffects:1024879): easyeffects-DEBUG: 20:08:34.862:         node_info_holder.cpp:104        240, Firefox finalized
Killed
```

**Comment by wwmm** [comment]
>Is it possible somehow to disable autokill ?

Maybe. I never tried to disable the `oom killer`.

>It feels that kill come in moment node_info_holder.cpp:104

Hum... Maybe it just happens to be the last print before whatever is actually triggering the issue. Even if the delete that comes after https://github.com/wwmm/easyeffects/blob/9471a39faf18733a00be4e020fb522c08ad28b35/src/node_info_holder.cpp#L104 has issues I would expect a segmentation fault or a double free instead of the system killing the process.

**Comment by wwmm** [comment]
> Hi . I have a new findings. In installed bcc package and used tool killsnoop -s 9 Appeared that kill signal not detected with killsnoop. So it means that most probabaly was used not a kill but tgkill syscall. I not sure about oom or segv. It is just direct non motivated kill signal. Issues started to manifest itself after last arch upgrade .

Hum... I did not consider that the kill signal could have different sources... Are you using the stable repository package or the AUR package? I do not remember if you have already given this information. It may be worth to install the AUR package if you have not tried that yet.

**Comment by jpVm5jYYRE1VIKL** [comment]
Did extra debug using PIPEWIRE_DEBUG=3 easyeffects 
and in moment of crash got following on console: 

```
I][20176.723590] pw.context   | [       context.c:  701 acquire_data_loop()] 0x5636ed359690: using name:'data-loop.0' class:'data.rt' ref:37
[I][20176.723611] pw.filter    | [        filter.c:  803 handle_latency()] port 0x5636ed2ca310: set input latency 1.000000-1.000000 0-0 0-0
[I][20176.723634] pw.node      | [     impl-node.c:  469 node_update_state()] (ee_soe_spectrum-101) creating -> running
[I][20176.724183] pw.node      | [     impl-node.c:  469 node_update_state()] (ee_soe_output_level-121) creating -> running
[I][20176.725122] pw.filter    | [        filter.c:  803 handle_latency()] port 0x5636ed5ff7d0: set output latency 1.000000-1.000000 0-0 0-0
[I][20176.725292] pw.context   | [       context.c:  725 pw_context_acquire_loop()] 0x5636ed359690: looking for name:'data-loop.0' class:'(null)'
[I][20176.725296] pw.context   | [       context.c:  701 acquire_data_loop()] 0x5636ed359690: using name:'data-loop.0' class:'data.rt' ref:38
[I][20176.725310] pw.filter    | [        filter.c:  803 handle_latency()] port 0x5636ed5f9c10: set output latency 0.000000-1.000000 0-0 0-0
[I][20176.725420] pw.filter    | [        filter.c:  803 handle_latency()] port 0x5636ed5fae40: set output latency 0.000000-1.000000 0-0 0-0
[I][20176.725639] pw.context   | [       context.c:  725 pw_context_acquire_loop()] 0x5636ed359690: looking for name:'data-loop.0' class:'(null)'
[I][20176.725643] pw.context   | [       context.c:  701 acquire_data_loop()] 0x5636ed359690: using name:'data-loop.0' class:'data.rt' ref:39
[I][20176.725666] pw.context   | [       context.c:  725 pw_context_acquire_loop()] 0x5636ed359690: looking for name:'data-loop.0' class:'(null)'
[I][20176.725670] pw.context   | [       context.c:  701 acquire_data_loop()] 0x5636ed359690: using name:'data-loop.0' class:'data.rt' ref:40
[I][20176.741195] spa.loop     | [          loop.c:  213 loop_create_queue()] 0x5636ed31e838 created queue 0x7fb5b01e8b90
[I][21236.752266] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.768191] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.781526] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.795406] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.806181] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.817225] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 2 wakeups
[I][21236.824736] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.832366] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.840136] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.848043] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.856092] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.864273] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.872601] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.881070] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.885357] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.894017] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.902815] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.907274] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.916277] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.920838] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.925435] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.934726] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.939432] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.944166] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.953717] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
[I][21236.958554] pw.node      | [     impl-node.c: 1548 node_on_fd_events()] (ee_soe_autogain-154) client missed 1 wakeups
Killed
```
And this is output on journald : 

```
Oct 26 08:01:02 ktotoPC dbus-broker[1127]: Peer :1.4835 is being disconnected as it does not have the resources to receive a reply it requested.
Oct 26 08:01:02 ktotoPC steam[2009185]: Adding process 2010808 for gameID 275850
Oct 26 08:01:02 ktotoPC steam[2009185]: Adding process 2010811 for gameID 275850
Oct 26 08:01:02 ktotoPC steam[2009185]: Adding process 2010822 for gameID 275850
Oct 26 08:01:03 ktotoPC steam[2009185]: Adding process 2010836 for gameID 275850
Oct 26 08:01:03 ktotoPC steam[2009185]: Adding process 2010842 for gameID 275850
Oct 26 08:01:03 ktotoPC steam[2009185]: Adding process 2010848 for gameID 275850
Oct 26 08:01:04 ktotoPC steam[2009185]: Adding process 2010861 for gameID 275850
Oct 26 08:01:04 ktotoPC steam[2009185]: Setting breakpad minidump AppID = 275850
Oct 26 08:01:04 ktotoPC steam[2009185]: Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198059180495 [API loaded no]
Oct 26 08:01:04 ktotoPC steam[2009185]: Adding process 2010877 for gameID 275850
Oct 26 08:01:11 ktotoPC kded6[1833]: Registering ":1.4408/StatusNotifierItem" to system tray
Oct 26 08:01:11 ktotoPC kded6[1833]: Service  ":1.4408" unregistered
Oct 26 08:01:11 ktotoPC wireplumber[1808232]: wp-event-dispatcher: <WpAsyncEventHook:0x5572b9ec69d0> failed: <WpSiStandardLink:0x5572ba2a3640> link failed: some node was destroyed before the link was created
Oct 26 08:01:11 ktotoPC plasmashell[1855]: QFont::setPointSizeF: Point size <= 0 (0.000000), must be greater than 0
Oct 26 08:01:11 ktotoPC kded6[1833]: Registering ":1.4409/StatusNotifierItem" to system tray
Oct 26 08:01:11 ktotoPC kded6[1833]: Service  ":1.4409" unregistered
Oct 26 08:01:11 ktotoPC kded6[1833]: Registering ":1.4410/StatusNotifierItem" to system tray
Oct 26 08:01:11 ktotoPC wireplumber[1808232]: wp-event-dispatcher: <WpAsyncEventHook:0x5572b9ec69d0> failed: <WpSiStandardLink:0x5572ba18df50> link failed: some node was destroyed before the link was created
Oct 26 08:01:11 ktotoPC kded6[1833]: Service  ":1.4410" unregistered
Oct 26 08:01:11 ktotoPC kded6[1833]: Registering ":1.4411/StatusNotifierItem" to system tray
Oct 26 08:01:11 ktotoPC kded6[1833]: Service  ":1.4411" unregistered
Oct 26 08:01:11 ktotoPC plasmashell[1855]: QFont::setPointSizeF: Point size <= 0 (0.000000), must be greater than 0
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 2 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups
Oct 26 08:01:22 ktotoPC easyeffects[2008355]: pw.node: (ee_soe_autogain-154) client missed 1 wakeups

```

**Comment by wwmm** [comment]
As far as I remember the node missing 1 or 2 wakeup should not cause the system to kill the process. Most installations out there probably also have 1 or 2 missed wakeups from time to time. It depends on the system load and on what pipewire is doing internally. As well as on as heavy the plugin is. The libebur128 library used in the autogain can be very CPU intensive compared to the other libraries we use.

**Comment by glanduin** [comment]
I don't know if related but i also am on Manjaro and in my case, easyeffects is killed when i introduce an equalizer (dependent on lsp-plugins package) and start playing audio. It could be any audio from any source, immediately kills easyeffects. If i remove the equalizer, everything works as expected. I also have "bass enhancer" and "stereo tools" (both provided by calf) in my effects pipeline and they work without issues. EQ (lsp) causes the crash of the whole app.

**Comment by wwmm** [comment]
> I don't know if related but i also am on Manjaro and in my case, easyeffects is killed when i introduce an equalizer (dependent on lsp-plugins package) and start playing audio. It could be any audio from any source, immediately kills easyeffects. If i remove the equalizer, everything works as expected. I also have "bass enhancer" and "stereo tools" (both provided by calf) in my effects pipeline and they work without issues. EQ (lsp) causes the crash of the whole app.

@glanduin which parameters have you changed in the equalizer configuration?

**Comment by glanduin** [comment]
I did not manually change anything but instead, i downloaded a "preset" from [https://autoeq.app/](https://autoeq.app/)

This preset is made spesifically for easyeffects and i loaded it as "APO" (as specified). It loaded succesfuly and i can see the eq parameters changed. I am suspecting some kind of clash between the LSP plugin package and pipewire maybe. Because this doesn't happen with CALF plugins. But i honestly did not test any other plugin from that package, so i am probably wrong.

**Comment by wwmm** [comment]
> I did not manually change anything but instead, i downloaded a "preset" from https://autoeq.app/
> 
> This preset is made spesifically for easyeffects and i loaded it as "APO" (as specified). It loaded succesfuly and i can see the eq parameters changed. I am suspecting some kind of clash between the LSP plugin package and pipewire maybe. Because this doesn't happen with CALF plugins. But i honestly did not test any other plugin from the LSP package, so i am probably wrong.

Most likely the preset changed only the number of bands and the corresponding settings without putting the equalizer into heavier modes of operations. So it should be fine.

Sometimes the system kills a process because it took too long to finish operations executed in a realtime thread. This may explain why only this plugin is causing problems for you. Do you see many errors in the output of `pw-top`?

**Comment by jpVm5jYYRE1VIKL** [comment]
I did this configuration : 
in ~/.config/pipewire/pipewire.conf.d
and it is quite improved situation but still sometimes killed. I started to look on pw-top and found that some applications ignoring min-quantum and attempt to use quantum less than 1024. Do exist any way to force to use bigger quantums ? 


```
context.properties = {
    default.clock.min-quantum = 1024
    default.clock.max-quantum = 8192
    default.clock.quantum = 1024
}
```

**Comment by wwmm** [comment]
>I started to look on pw-top and found that some applications ignoring min-quantum and attempt to use quantum less than 1024. Do exist any way to force to use bigger quantums ?

The numbers you are seeing just represent the requested latency. PipeWire may or not switch to them. The line that matters is the one for your soundcard. This is the quantum PipeWire will use.

>PS. And yes i almost sure that problem with autogain plugin because in pw-top i see most of errors on ee_soe_autogain.
Is it possible to improve autogain plugin performance because it is quite important and i cannot survive without it.

The main reason why this plugin is heavy is that the library `libebur128` used to estimate the loudness values is heavy. Unless this library somehow becomes faster the only way to make the autogain to use less CPU is forcing a higher latency (quantum) value. This will make PipeWire to deliver larger buffer sizes. What in turns usually decreases the number of times the library functions have to be called.

@jpVm5jYYRE1VIKL have you ever installed ananicy or something else that could be messing with EasyEffects threads priorities? Run the command `chrt -ap easyeffects_pid_number`. Only one of EasyEffects threads should have realtime priority. Setting the whole app to realtime is a common reason for the system to kill it.

**Comment by jpVm5jYYRE1VIKL** [comment]
hi. @wwmm 
Pririties is good.
Issue with pw-top ERR extremely improved after already some hours i don`t have even 1 ERR in pw-top. 
To improve it i used  in
**/etc/modprobe.d/blacklist.conf** 

`install snd_hda_codec_hdmi /bin/true`

This action blocked HDMI sound destination . I have multiply monitors but not using monitors output for sound processing. May be it will be good to add faq advice to block extra sound outputs even if it is not used. As i see it indirectly influence on EE even if not used and even if sound never directed to such output. May be nvidia driver ? I ll test more and report. At the moment, the situation with hdmi block is very encouraging.

**Comment by wwmm** [comment]
> This action blocked HDMI sound destination . I have multiply monitors but not using monitors output for sound processing. May be it will be good to add faq advice to block extra sound outputs even if it is not used. As i see it indirectly influence on EE even if not used and even if sound never directed to such output. May be nvidia driver ? I ll test more and report. At the moment, the situation with hdmi block is very encouraging.

Sounds like a bug in the driver or in how PipeWire is handling your devices. Besides my onboard card I also have my monitor audio hdmi device enabled without any side effect.

**Comment by jpVm5jYYRE1VIKL** [comment]
After couple  days of testing i can say that EE still killed but situation dramatically improved. 
Issues happens not during playback itself. issue happens in moments playback->pause-> playback. 
After HDMI block i got only one crash during 2 days. Before it was multiply times per day. I started to think about last intel CPU instability issue but i am not sure about it because only application which crashing is EE. Quite big and cpu hungry games never crash.

**Comment by wwmm** [comment]
>but i am not sure about it because only application which crashing is EE. Quite big and cpu hungry games never crash.

The difference is probably the realtime thread EasyEffects uses for the plugins audio processing. Maybe somehow the CPU is not being fast enough to initialize the plugin within the time PipeWire configures for realtime threads. But as far as I remember PipeWire sets a really large timeout.

**Comment by glanduin** [comment]
I suggest installing and trying the flatpak version of easy effect. It solved my problems. It comes with all the plugins and works perfectly fine on my Manjaro ARM.

**Comment by jpVm5jYYRE1VIKL** [comment]
i completely solved issue using following workaround : 
i disabled Use default Input and Use default Output . 
This way this issue possible to solve only if your EE killed in moments of pause/playback pressing button or loading new apps which require sound. 

Now i completely sure that issue observed do not have any relation to CPU . Issue related to reconnection of pipewire nodes in moment if new app appear and pipewire node links reestablish and it crash really only in a moment when pipewire link reestablish happens. it is quite visitble if to load app like "qpwgraph" . All KIlls happens not in moment when playback happens but in moment what you press pause of playback . 

PS: Beside this after . Be aware that EE if in Device management not properly return settings after wakeup of computer. And after you wakeup PC you have to reassign manually output device again  .

**Comment by jpVm5jYYRE1VIKL** [comment]
After long investigations i found that easy effects die in case if intalled and loaded pipewire-zeroconf libpipewire-module-raop-discover. EE die even if RAOP not used but loaded.

**Comment by wwmm** [comment]
> After long investigations i found that easy effects die in case if intalled and loaded pipewire-zeroconf libpipewire-module-raop-discover. EE die even if RAOP not used but loaded.

Interesting. I think I have never loaded any of these modules on my computer. I wonder if pipewire is putting them in the same realtime thread as our filters.

**Comment by jpVm5jYYRE1VIKL** [comment]
This is what happened in a moment of EE death  sigkill was generated by   data-loop.0 : 

```
     data-loop.0-286427  [026] d..1. 147482.949467: signal_generate: sig=9 errno=0 code=128 comm=data-loop.0 pid=286427 grp=1 res=0
           <...>-286422  [014] d..1. 147482.949468: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286420  [019] d..1. 147482.949469: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286418  [022] d..1. 147482.949470: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286443  [000] d..1. 147482.949471: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286440  [017] d..1. 147482.949472: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
     data-loop.0-286427  [026] d..1. 147482.949473: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286442  [004] d..1. 147482.949474: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286422  [014] ..... 147482.949477: sched_process_exit: comm=[pango] fontcon pid=286422 prio=120 group_dead=false
           <...>-286420  [019] ..... 147482.949477: sched_process_exit: comm=dconf worker pid=286420 prio=120 group_dead=false
           <...>-286418  [022] ..... 147482.949478: sched_process_exit: comm=gmain pid=286418 prio=120 group_dead=false
           <...>-286443  [000] ..... 147482.949478: sched_process_exit: comm=[vkps] Update pid=286443 prio=120 group_dead=false
           <...>-286440  [017] ..... 147482.949478: sched_process_exit: comm=[vkcf] Analysis pid=286440 prio=120 group_dead=false
     data-loop.0-286427  [026] ..... 147482.949478: sched_process_exit: comm=data-loop.0 pid=286427 prio=79 group_dead=false
           <...>-286442  [004] ..... 147482.949480: sched_process_exit: comm=[vkrt] Analysis pid=286442 prio=120 group_dead=false
     easyeffects-286445  [002] d..1. 147482.949481: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
     easyeffects-286445  [002] ..... 147482.949486: sched_process_exit: comm=easyeffects pid=286445 prio=120 group_dead=false
           <...>-286419  [024] d..1. 147482.949504: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286419  [024] ..... 147482.949505: sched_process_exit: comm=pool-spawner pid=286419 prio=120 group_dead=false
           <...>-286417  [008] d..1. 147482.949542: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286426  [021] d..1. 147482.949546: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286425  [025] d..1. 147482.949546: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286426  [021] ..... 147482.949547: sched_process_exit: comm=module-rt pid=286426 prio=120 group_dead=false
           <...>-286425  [025] ..... 147482.949548: sched_process_exit: comm=ee-pipewire-thr pid=286425 prio=120 group_dead=false
           <...>-286421  [027] d..1. 147482.949549: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286421  [027] ..... 147482.949550: sched_process_exit: comm=gdbus pid=286421 prio=120 group_dead=false
           <...>-286417  [008] ..... 147482.949552: sched_process_exit: comm=easyeffects pid=286417 prio=120 group_dead=false
           <...>-286439  [010] d..1. 147482.949579: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286472  [012] d..1. 147482.949582: signal_deliver: sig=9 errno=0 code=0 sa_handler=0 sa_flags=0
           <...>-286472  [012] ..... 147482.949585: sched_process_exit: comm=data-loop.0 pid=286472 prio=120 group_dead=false
           <...>-286439  [010] ..... 147482.949656: sched_process_exit: comm=easyeffects pid=286439 prio=120 group_dead=true
           <...>-286438  [025] ..... 147482.953808: sched_process_exit: comm=nvidia-drm/time pid=286438 prio=120 group_dead=true
           <...>-286441  [018] ..... 147482.953850: sched_process_exit: comm=nvidia-drm/time pid=286441 prio=120 group_dead=true
           <...>-286444  [002] ..... 147482.953870: sched_process_exit: comm=nvidia-drm/time pid=286444 prio=120 group_dead=true
           <...>-286416  [010] d..1. 147483.004519: signal_deliver: sig=17 errno=0 code=2 sa_handler=557a42d06df0 sa_flags=14000000
           <...>-286416  [010] ..... 147483.004562: sched_process_exit: comm=bash pid=286416 prio=120 group_dead=true
           <...>-284646  [014] d..1. 147483.004679: signal_deliver: sig=17 errno=0 code=1 sa_handler=557a42d06df0 sa_flags=14000000
           <...>-291882  [025] d..1. 147483.107239: signal_deliver: sig=31 errno=24 code=1 sa_handler=7fd92bf98f60 sa_flags=44000004
```

i found a way how to easy recreate issue : 

this is kstack in moment of kill : 

2025-10-25 08:07:46 SIGKILL to pid=296668 comm=data-loop.0 code=128
--- kstack ---

        __traceiter_signal_generate+74
        __traceiter_signal_generate+74
        __send_signal_locked+628
        posix_cpu_timers_work+366
        task_work_run+90
        irqentry_exit_to_user_mode+552
        asm_sysvec_apic_timer_interrupt+26

2025-10-25 08:10:55 SIGKILL to pid=316733 comm=data-loop.0 code=128
--- kstack ---

        __traceiter_signal_generate+74
        __traceiter_signal_generate+74
        __send_signal_locked+628
        posix_cpu_timers_work+366
        task_work_run+90
        irqentry_exit_to_user_mode+552
        asm_sysvec_apic_timer_interrupt+26

Checked Limits : 

systemctl --user show easyeffects.service | grep -i LimitCPU
LimitCPU=infinity
LimitCPUSoft=infinity

**Comment by jpVm5jYYRE1VIKL** [comment]
prlimit --rttime=unlimited --pid EE_pid solve issue

Issue is because easyeffect during start ignoring 
systemctl --user show easyeffects.service | grep LimitRTT

LimitRTTIME=infinity
LimitRTTIMESoft=infinity
 
but after start i see : 

prlimit --pid 324702 --rttime
RESOURCE DESCRIPTION                   SOFT   HARD UNITS
RTTIME   timeout for real-time tasks 200000 200000 microsecs

And forced to do manually after EE start : prlimit --rttime=unlimited --pid EE_pid to solve issue

**Comment by wwmm** [comment]
> prlimit --rttime=unlimited --pid EE_pid solve issue
> 
> Issue is because easyeffect during start ignoring systemctl --user show easyeffects.service | grep LimitRTT
> 
> LimitRTTIME=infinity LimitRTTIMESoft=infinity
> 
> but after start i see :
> 
> prlimit --pid 324702 --rttime RESOURCE DESCRIPTION SOFT HARD UNITS RTTIME timeout for real-time tasks 200000 200000 microsecs
> 
> And forced to do manually after EE start : prlimit --rttime=unlimited --pid EE_pid to solve issue

If something like this is only necessary when `pipewire-zeroconf` and `libpipewire-module-raop-discover` are loaded it might be the case of reporting this situation to pipewire's developers.

**Comment by jpVm5jYYRE1VIKL** [comment]
@wwmm not really it is 100% EE issue. I found a way to recreate the issue even on a clean system. Because I found this way, it was quite easy to see that the kill signal was generated by the kernel. After that, I started to look into why the kernel can kill a process and got a lot of advice on the Internet that it can be related to OOM. I did a lot of tests to exclude OOM, and I was also confused when I looked at the D-Bus systemd service unit configuration.

But after I found that prlimit showed that in reality the RTT timeout on process EE is 200 microseconds instead of infinity, the puzzle was solved. So now the only question is why EE is ignoring LimitRTTIME. Is it possible that somewhere it is hardcoded to 200 microseconds (a call to setrlimit from EE???).

Even if I manually set it to 400 microseconds, the issue is completely solved. Now I want to automate it, but the problem is that prlimit requires sudo. Besides this, I see a lot of people for months complaining about killed EE processes without any reason. Most probably, all such issues are related to RT time.


PS : 

Then I executed the command:
strace -ff -o strace.out -e prlimit64 -- easyeffects

and one of the child threads of EasyEffects executed the following call:

prlimit64(0, RLIMIT_RTTIME, {rlim_cur=200000, rlim_max=200000}, NULL) = 0

**Comment by wwmm** [comment]
>So now the only question is why EE is ignoring LimitRTTIME

EasyEffects does not touch anything related to the realtime thread. Its management is entirely in PipeWire's hands. What we try is being as fast as possible when processing audio buffers. But depending on the algorithm it is what it is. And most of the plugins we use are from other projects. So even if it is possible to make them faster it is out of my reach to do it.

That being said I have seen pw-top report in some situations busy values a lot higher than 200 microseconds on my systems and this did not make EasyEffects be killed here.

>Is it possible that somewhere it is hardcoded to 200 microseconds (a call to setrlimit from EE???)

PipeWire is the one setting the limits. But as far as I remember its limits were quite generous. A few milliseconds if I am not mistaken. Maybe they've changed that.

**Comment by jpVm5jYYRE1VIKL** [comment]
> That being said I have seen pw-top report in some situations busy values a lot higher than 200 microseconds on my systems and this did not make EasyEffects be killed here.

I also did a test during quite big load and everything worked fine. Issue happens in a moments of switching start-stop-start-stop.  Sorry i am not a programmer and i not understand exactly how it is works. But for me is look weird how pipewire able to influense and set RTlimits for threads in process grtoup of EE. data-loop started and managed by EE or by Pipewire ? In reality in process group EE i see two data-loops.

**Comment by wwmm** [comment]
>But for me is look weird how pipewire able to influense and set RTlimits for threads in process grtoup of EE. data-loop started and managed by EE or by Pipewire ? 

PipeWire creates the realtime thread and invokes the plugin `on_process` callback we define in this thread. The whole flow of audio buffers in the effects pipeline is controlled by pipewire. Linking or unlinking the filters in some situations is our task. But the one deciding when the audio data comes and on which thread the data process function runs is pipewire.
